src/operations/list_collections.ts

```diff
@@ -47,17 +48,16 @@ export class ListCollectionsOperation extends CommandCallbackOperation<string[]>
     }
   }
 
-  override executeCallback(
-    server: Server,
-    session: ClientSession | undefined,
-    callback: Callback<string[]>
-  ): void {
-    return super.executeCommandCallback(
+  override execute(server: Server, session: ClientSession | undefined): Promise<string[]> {
+    return super.executeCommand(
       server,
       session,
-      this.generateCommand(maxWireVersion(server)),
-      callback
-    );
+      this.generateCommand(maxWireVersion(server))
+    ) as TODO_NODE_3286;
```

Comment:
The TODO does not apply here, the generic is just incorrect. (Sorry 😅). I know this because MongoDB cannot return arrays as the top-level result. 

Hop in the debugger to confirm the result of this operation. This is a cursor creating command so the result is going to be something like: `{cursor:{id: number, firstBatch: Document[]}}`, (not sure the exact shape) 

It would be more correct to use `Document` at the very least, or we can declare a new `@internal` type `ListCollectionsResult` that can more accurately capture the expected response.